Lib/test/test_dbm_gnu.py

```diff
@@ -205,6 +206,15 @@ def test_clear(self):
                 self.assertNotIn(k, db)
             self.assertEqual(len(db), 0)
 
+    @support.run_with_locale(
+        'LC_ALL',
+        'fr_FR.iso88591', 'ja_JP.sjis', 'zh_CN.gbk',
+        '',
+    )
+    def test_localized_error(self):
+        with temp_dir() as d:
+            create_empty_file(os.path.join(d, TESTFN))
```

Comment:
TESTFN is non-ASCII which is encodable with the filesystem encoding when tests are imported. It may be not encodable in the specified test locale. So I suggest to use name containing `'\udcff'` or something like.

It may be impossible to create a non-UTF-8 file name on macOS, so catch an error in `create_empty_file()` and skip the test.

You may need to test this on Windows manually to ensure that the test works and not always skipped (if gdbm is available on Windows).

Reply:
Ah I thought it would have been file. My question: since I don't care about the filename in this case (the error doesn't report it), I can just use an ASCII name and that's it right? (I'm just interested in not having UnicodeDecodeError)